### ISSUE TITLE
Did anyone face any issue like this?

### ISSUE THREAD

**Issue by abhi3700** [uuosio/ipyeos#2]
**Context**
I was trying the hello tutorial from the [book](https://uuosio.github.io/rscdk-book/helloworld-source-code.html). And during the testing, I found this error:

```
❯ eosdebugger                                                                                                                             ⏎
export CHAIN_API_LIB=/Users/abhi3700/opt/anaconda3/lib/python3.9/site-packages/ipyeos/release/lib/libchain_api.dylib
export VM_API_LIB=/Users/abhi3700/opt/anaconda3/lib/python3.9/site-packages/ipyeos/release/lib/libvm_api.dylib
export PYTHON_SHARED_LIB_PATH=/Users/abhi3700/opt/anaconda3/lib/libpython3.9.dylib
/Users/abhi3700/opt/anaconda3/lib/python3.9/site-packages/ipyeos/release/bin/ipyeos -m ipyeos eosdebugger
dyld[10134]: Library not loaded: /usr/local/opt/zstd/lib/libzstd.1.dylib
  Referenced from: <9D771806-EEBF-3787-9792-2F1465279363> /Users/abhi3700/opt/anaconda3/lib/python3.9/site-packages/ipyeos/release/bin/ipyeos
  Reason: tried: '/usr/local/opt/zstd/lib/libzstd.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/usr/local/opt/zstd/lib/libzstd.1.dylib' (no such file), '/usr/local/opt/zstd/lib/libzstd.1.dylib' (no such file), '/usr/local/lib/libzstd.1.dylib' (no such file), '/usr/lib/libzstd.1.dylib' (no such file, not in dyld cache)
```

I have tried installing `zstd` using `brew install zstd`, still same error.

Any help would be really appreciated.
Thanks 🙏

**Comment by ghost** [comment]
`brew install zstd` works for me. 

What version of macOS are you using?

Could you check if `/usr/local/opt/zstd/lib/libzstd.1.dylib` exists after the installation?

Also, you can use python 3.10 to install ipyeos in order to avoid this issue.

**Comment by abhi3700** [comment]
@learnforpractice 
- **macOS**: 13.2
- Yes, it's already there.
<img width="292" alt="image" src="https://user-images.githubusercontent.com/16472948/215959965-aa3670be-3fc0-4a8f-9f81-4dbe954b5e8e.png">


- I used `python3.10`, but it throws error like this:
<img width="1190" alt="image" src="https://user-images.githubusercontent.com/16472948/215959748-057c2b7f-37e1-4d81-91dd-1d6b95a8e1d1.png">

I checked for missing module, so I installed `scikit-build`, `opencv-python` from [here](https://stackoverflow.com/questions/63448467/installing-opencv-fails-because-it-cannot-find-skbuild). Still same issue.

**Comment by ghost** [comment]
Try this command:

```
python3.10 -m pip install -U ipyeos
```

**Comment by abhi3700** [comment]
> Try this command:
> 
> ```
> python3.10 -m pip install -U ipyeos
> ```

same issue as previous:

<img width="1152" alt="image" src="https://user-images.githubusercontent.com/16472948/216048433-b6609840-8043-48e7-baa9-14b09b1cd3f5.png">

**Comment by ghost** [comment]
Strange. Are you using a MacBook Pro with M1 chip? Currently, `ipyeos` does not support M1 chip. In this case, I suggest using `docker` to run `eosdebugger`

```
docker pull ghcr.io/uuosio/ipyeos:latest
docker run -it --rm -p 9090:9090 -p 9092:9092 -t ghcr.io/uuosio/ipyeos
```

**Comment by abhi3700** [comment]
Yes! I am using M1 chip.

> `ipyeos` does not support M1 chip
Understood 😞 

I did try, it seems there is any issue from the docker image:
```
$ docker pull ghcr.io/uuosio/ipyeos:latest
Error response from daemon: error creating temporary lease: connection error: desc = "transport: Error while dialing dial unix:///run/containerd/containerd.sock: timeout": unavailable
```
<img width="1279" alt="image" src="https://user-images.githubusercontent.com/16472948/216084231-280ee124-0b7e-419c-b78e-dc7eaf34a869.png">

**Comment by ghost** [comment]
As I don't have access to an M1 machine, I did a web search to gather information that might assist with your issue.

Here are a few relevant resources I found: 

https://docs.docker.com/desktop/install/mac-install

https://stackoverflow.com/questions/69054921/docker-on-mac-m1-gives-the-requested-images-platform-linux-amd64-does-not-m

https://collabnix.com/choosing-the-right-docker-image-for-your-apple-m1-pro

To summarize, the following steps may resolve the issue:

1. Install Docker Desktop for Apple silicon
2. Launch the container using the command: `docker run --platform linux/amd64 -it --rm -p 9090:9090 -p 9092:9092 -t ghcr.io/uuosio/ipyeos`

**Comment by abhi3700** [comment]
> 1. Install Docker Desktop for Apple silicon
<img width="1138" alt="image" src="https://user-images.githubusercontent.com/16472948/216360592-1bec0a3d-9f70-471f-a232-499282786e59.png">

> 2. Launch the container using the command: `docker run --platform linux/amd64 -it --rm -p 9090:9090 -p 9092:9092 -t ghcr.io/uuosio/ipyeos`
<img width="933" alt="image" src="https://user-images.githubusercontent.com/16472948/216361184-46317521-6d4f-4514-9df8-0a43289a3414.png">

Still same problem.
